In [28]:
import pandas as pd 

In [29]:
df_basic = pd.read_csv('survey_results_public.csv', encoding='utf-8',low_memory=False)

In [30]:
df_basic.info(memory_usage='deep')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49191 entries, 0 to 49190
Columns: 172 entries, ResponseId to JobSat
dtypes: float64(52), int64(1), object(119)
memory usage: 357.8 MB


In [31]:
pd.set_option('display.max_columns', None)

In [32]:
df_basic.head()

,ResponseId,MainBranch,Age,EdLevel,Employment,EmploymentAddl,WorkExp,LearnCodeChoose,LearnCode,LearnCodeAI,AILearnHow,YearsCode,DevType,OrgSize,ICorPM,RemoteWork,PurchaseInfluence,TechEndorseIntro,TechEndorse_1,TechEndorse_2,TechEndorse_3,TechEndorse_4,TechEndorse_5,TechEndorse_6,TechEndorse_7,TechEndorse_8,TechEndorse_9,TechEndorse_13,TechEndorse_13_TEXT,TechOppose_1,TechOppose_2,TechOppose_3,TechOppose_5,TechOppose_7,TechOppose_9,TechOppose_11,TechOppose_13,TechOppose_16,TechOppose_15,TechOppose_15_TEXT,Industry,JobSatPoints_1,JobSatPoints_2,JobSatPoints_3,JobSatPoints_4,JobSatPoints_5,JobSatPoints_6,JobSatPoints_7,JobSatPoints_8,JobSatPoints_9,JobSatPoints_10,JobSatPoints_11,JobSatPoints_13,JobSatPoints_14,JobSatPoints_15,JobSatPoints_16,JobSatPoints_15_TEXT,AIThreat,NewRole,ToolCountWork,ToolCountPersonal,Country,Currency,CompTotal,LanguageChoice,LanguageHaveWorkedWith,LanguageWantToWorkWith,LanguageAdmired,LanguagesHaveEntry,LanguagesWantEntry,DatabaseChoice,DatabaseHaveWorkedWith,DatabaseWantToWorkWith,DatabaseAdmired,DatabaseHaveEntry,DatabaseWantEntry,PlatformChoice,PlatformHaveWorkedWith,PlatformWantToWorkWith,PlatformAdmired,PlatformHaveEntry,PlatformWantEntry,WebframeChoice,WebframeHaveWorkedWith,WebframeWantToWorkWith,WebframeAdmired,WebframeHaveEntry,WebframeWantEntry,DevEnvsChoice,DevEnvsHaveWorkedWith,DevEnvsWantToWorkWith,DevEnvsAdmired,DevEnvHaveEntry,DevEnvWantEntry,SOTagsHaveWorkedWith,SOTagsWantToWorkWith,SOTagsAdmired,SOTagsHaveEntry,SOTagsWant Entry,OpSysPersonal use,OpSysProfessional use,OfficeStackAsyncHaveWorkedWith,OfficeStackAsyncWantToWorkWith,OfficeStackAsyncAdmired,OfficeStackHaveEntry,OfficeStackWantEntry,CommPlatformHaveWorkedWith,CommPlatformWantToWorkWith,CommPlatformAdmired,CommPlatformHaveEntr,CommPlatformWantEntr,AIModelsChoice,AIModelsHaveWorkedWith,AIModelsWantToWorkWith,AIModelsAdmired,AIModelsHaveEntry,AIModelsWantEntry,SOAccount,SOVisitFreq,SODuration,SOPartFreq,SO_Dev_Content,SO_Actions_1,SO_Actions_16,SO_Actions_3,SO_Actions_4,SO_Actions_5,SO_Actions_6,SO_Actions_9,SO_Actions_7,SO_Actions_10,SO_Actions_15,SO_Actions_15_TEXT,SOComm,SOFriction,AISelect,AISent,AIAcc,AIComplex,AIToolCurrently partially AI,AIToolDon't plan to use AI for this task,AIToolPlan to partially use AI,AIToolPlan to mostly use AI,AIToolCurrently mostly AI,AIFrustration,AIExplain,AIAgents,AIAgentChange,AIAgent_Uses,AgentUsesGeneral,AIAgentImpactSomewhat agree,AIAgentImpactNeutral,AIAgentImpactSomewhat disagree,AIAgentImpactStrongly agree,AIAgentImpactStrongly disagree,AIAgentChallengesNeutral,AIAgentChallengesSomewhat disagree,AIAgentChallengesStrongly agree,AIAgentChallengesSomewhat agree,AIAgentChallengesStrongly disagree,AIAgentKnowledge,AIAgentKnowWrite,AIAgentOrchestration,AIAgentOrchWrite,AIAgentObserveSecure,AIAgentObsWrite,AIAgentExternal,AIAgentExtWrite,AIHuman,AIOpen,ConvertedCompYearly,JobSat
0,1,I am a developer by profession,25-34 years old,"Master’s degree (M.A., M.S., M.Eng., MBA, etc.)",Employed,"Caring for dependents (children, elderly, etc.)",8.0,"Yes, I am not new to coding but am learning ne...",Online Courses or Certification (includes all ...,"Yes, I learned how to use AI-enabled tools for...",AI CodeGen tools or AI-enabled apps,14.0,"Developer, mobile",20 to 99 employees,People manager,Remote,"Yes, I influenced the purchase of a substantia...",Work,10.0,7.0,9.0,6.0,3.0,11.0,12.0,1.0,8.0,14.0,NaN,15.0,7.0,8.0,12.0,11.0,1.0,6.0,13.0,3.0,16.0,NaN,Fintech,3.0,1.0,4.0,9.0,5.0,10.0,12.0,11.0,2.0,6.0,7.0,13.0,14.0,15.0,8.0,NaN,I'm not sure,I have neither consider or transitioned into a...,7.0,3.0,Ukraine,EUR European Euro,52800.0,Yes,Bash/Shell (all shells);Dart;SQL,Dart,Dart,NaN,NaN,Yes,Cloud Firestore;PostgreSQL,NaN,NaN,NaN,NaN,Yes,Amazon Web Services (AWS);Cloudflare;Firebase;...,NaN,NaN,NaN,NaN,No,NaN,NaN,NaN,NaN,NaN,Yes,Android Studio;Notepad++;Visual Studio;Visual ...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Windows;MacOS;Android,Windows;MacOS;Android;iOS;iPadOS,Confluence;GitHub;GitLab;Jira;Markdo

In [33]:
df_structure = pd.read_csv('survey_results_schema.csv')

In [34]:
df_structure.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 139 entries, 0 to 138
Data columns (total 6 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   qid       139 non-null    object 
 1   qname     139 non-null    object 
 2   question  139 non-null    object 
 3   type      139 non-null    object 
 4   sub       49 non-null     object 
 5   sq_id     49 non-null     float64
dtypes: float64(1), object(5)
memory usage: 6.6+ KB


In [35]:
df_structure.head()

,qid,qname,question,type,sub,sq_id
0,QID18,TechEndorse_1,What attracts you to a technology or causes yo...,RO,AI integration or AI Agent capabilities,1.0
1,QID18,TechEndorse_2,What attracts you to a technology or causes yo...,RO,Easy-to-use API,2.0
2,QID18,TechEndorse_3,What attracts you to a technology or causes yo...,RO,Robust and complete API,3.0
3,QID18,TechEndorse_4,What attracts you to a technology or causes yo...,RO,Customizable and manageable codebase,4.0
4,QID18,TechEndorse_5,What attracts you to a technology or causes yo...,RO,Reputation for quality,5.0


In [36]:
df_basic.describe().round(2)

,ResponseId,WorkExp,YearsCode,TechEndorse_1,TechEndorse_2,TechEndorse_3,TechEndorse_4,TechEndorse_5,TechEndorse_6,TechEndorse_7,TechEndorse_8,TechEndorse_9,TechEndorse_13,TechOppose_1,TechOppose_2,TechOppose_3,TechOppose_5,TechOppose_7,TechOppose_9,TechOppose_11,TechOppose_13,TechOppose_16,TechOppose_15,JobSatPoints_1,JobSatPoints_2,JobSatPoints_3,JobSatPoints_4,JobSatPoints_5,JobSatPoints_6,JobSatPoints_7,JobSatPoints_8,JobSatPoints_9,JobSatPoints_10,JobSatPoints_11,JobSatPoints_13,JobSatPoints_14,JobSatPoints_15,JobSatPoints_16,ToolCountWork,ToolCountPersonal,CompTotal,SO_Actions_1,SO_Actions_16,SO_Actions_3,SO_Actions_4,SO_Actions_5,SO_Actions_6,SO_Actions_9,SO_Actions_7,SO_Actions_10,SO_Actions_15,ConvertedCompYearly,JobSat
count,49191.00,42893.00,43042.00,35975.00,35975.00,35975.00,35975.00,35975.00,35975.00,35975.00,35975.00,35975.00,35975.00,34259.00,34259.00,34259.00,34259.00,34259.00,34259.0,34259.00,34259.00,34259.00,34259.00,24943.00,24943.00,24943.00,24943.00,24943.00,24943.00,24943.00,24943.00,24943.00,24943.00,24943.00,24943.00,24943.00,24943.00,24943.00,27611.00,25582.00,2.486600e+04,26260.00,24340.00,26260.00,26260.00,26260.00,26260.00,26260.00,26260.00,26260.00,26260.00,23947.00,26670.0
mean,24596.00,13.37,16.57,7.87,4.10,4.11,5.68,4.12,5.23,6.48,4.37,4.43,9.85,8.31,4.52,5.93,4.99,4.05,3.8,4.37,5.06,5.51,9.95,6.87,5.22,7.46,9.54,9.76,8.19,5.88,7.37,6.74,7.51,5.70,8.56,8.68,14.69,7.83,17.73,8.55,2.234198e+70,4.91,3.34,5.72,4.56,4.79,5.20,5.68,4.98,7.10,10.08,101761.54,7.2
std,14200.36,10.80,11.79,2.40,2.28,2.33,2.40,2.44,2.80,2.33,2.36,2.62,1.98,2.20,2.28,2.30,2.29,2.61,2.5,2.56,2.53,2.91,2.09,3.74,3.66,3.73,3.78,3.56,3.80,3.90,4.12,3.89,3.85,4.11,3.55,3.63,1.90,4.08,269.81,44.66,3.523096e+72,2.68,2.41,2.63,3.07,2.64,2.56,2.31,2.49,2.47,1.94,461756.90,2.0
min,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.0,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,0.00,0.00,0.000000e+00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,1.00,0.0
25%,12298.50,5.00,8.00,7.00,2.00,2.00,4.00,2.00,3.00,5.00,2.00,2.00,10.00,8.00,3.00,4.00,3.00,2.00,2.0,2.00,3.00,3.00,10.00,4.00,2.00,4.00,7.00,7.00,5.00,2.00,4.00,3.00,4.00,2.00,6.00,6.00,15.00,4.00,4.00,3.00,5.500000e+04,3.00,1.00,3.00,1.00,3.00,3.00,4.00,3.00,6.00,10.00,38171.00,6.0
50%,24596.00,10.00,14.00,9.00,4.00,4.00,6.00,4.00,5.00,7.00,4.00,4.00,10.00,9.00,4.00,6.00,5.00,4.00,3.0,4.00,5.00,6.00,10.00,6.00,4.00,7.00,10.00,10.00,8.00,5.00,7.00,6.00,7.00,5.00,9.00,9.00,15.00,8.00,6.00,5.00,1.050000e+05,5.00,2.00,6.00,4.00,5.00,5.00,6.00,5.00,8.00,10.00,75320.00,8.0
75%,36893.50,20.00,24.00,9.00,6.00,6.00,8.00,6.00,8.00,8.00,6.00,7.00,10.00,9.00,6.00,8.00,7.00,6.00,6.0,6.00,7.00,8.00,10.00,10.00,8.00,11.00,13.00,13.00,11.00,9.00,11.00,10.00,11.00,9.00,12.00,12.00,15.00,11.00,10.00,7.00,2.400000e+05,7.00,5.00,8.00,7.00,7.00,7.00,7.00,7.00,9.00,10.00,120596.00,8.0
max,49191.00,100.00,100.00,14.00,14.00,14.00,14.00,14.00,14.00,14.00,14.00,14.00,14.00,15.00,15.00,15.00,15.00,15.00,15.0,14.00,15.00,15.00,16.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,10000.00,1000.00,5.555556e+74,15.00,10.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,15.00,50000000.00,10.0


In [37]:
# whether there are no duplicates according to the unique number of the respondent.
dublicat_id = df_basic.duplicated('ResponseId').sum()
print(f'\n Number of duplicates = {dublicat_id}')


 Number of duplicates = 0


In [38]:
ResponseId_count=df_basic.shape[0]
print(f"\n {ResponseId_count}")


 49191


In [39]:
# We create a set of questions and immediately make an intersection
questions = set(df_structure['qname'].unique()) & set(df_basic.columns)
print(f"\n Common elements found: {len(questions)}")


 Common elements found: 126


In [40]:
# We convert the set into a list for Pandas indexing
questions_list = list(questions)

# Add ResponseId for identification
cols_to_keep = ['ResponseId'] + questions_list

# We create the final working dataframe
df_final = df_basic[cols_to_keep]

print(f"\n New table dimension: {df_final.shape}")


 New table dimension: (49191, 127)


In [41]:
# We remove only those who did not answer ANY of these questions
df_cleaned = df_final.dropna(subset=questions_list, how='all')
print(f"\ n size of the cleared table: {df_cleaned.shape}")

\ n size of the cleared table: (49191, 127)


In [42]:
# Count the number of respondents without missing values
df_without_missing = df_cleaned[questions_list].notna().all(axis=1)
respondents_no_nan = df_without_missing.sum()
print(f"\n Full answers:{respondents_no_nan}")



 Full answers:0


In [43]:
# Calculation of measures of central tendency
mean_exp = df_cleaned['WorkExp'].mean().round(1)
median_exp = df_cleaned['WorkExp'].median()
mode_exp = df_cleaned['WorkExp'].mode()[0]  # .mode() returns Series, we take the first element
# Outputting the result
print("\n Measures of Central Tendency for Work Experience (WorkExp):\n")
print(f" Mean    {mean_exp} ")
print(f" Median  {median_exp}")
print(f" Mode   {mode_exp}")


 Measures of Central Tendency for Work Experience (WorkExp):

 Mean    13.4 
 Median  10.0
 Mode   10.0


In [44]:
# We filter respondents who chose "Remote"
remote_workers = df_cleaned[df_cleaned['RemoteWork'] == 'Remote']
# We count their number
remote_count = remote_workers.shape[0]
# We derive the final result
print(f"\n The number of respondents who work remotely: {remote_count}")


 The number of respondents who work remotely: 10931


In [45]:
# We are looking for the name of the column that contains the word 'Language'
matching_cols = [col for col in df_basic.columns if 'Language' in col]
print("Found similar columns:", matching_cols)

Found similar columns: ['LanguageChoice', 'LanguageHaveWorkedWith', 'LanguageWantToWorkWith', 'LanguageAdmired', 'LanguagesHaveEntry', 'LanguagesWantEntry']


In [46]:
# Define the name of the column
lang_col = 'LanguageHaveWorkedWith'

# We filter those who specified Python. we use .str.contains, na=False ignores empty values ​​(NaN)
python_users = df_basic[df_basic[lang_col].str.contains('python', case=False, na=False)]

#  Counting the number of Python users
python_count = python_users.shape[0]

# We count the total number of respondents (for the denominator)
total_respondents = df_basic.shape[0]
# We calculate the percentage
python_percentage = (python_count / total_respondents) * 100

# We derive the result
print(f"\n Percentage of respondents who program in Python: {python_percentage:.1f}%")


 Percentage of respondents who program in Python: 37.5%


In [47]:
# We are looking for the name of the column that contains the word 'learn'
matching_cols = [col for col in df_basic.columns if 'Learn' in col]
print("\n Found similar columns:", matching_cols)


 Found similar columns: ['LearnCodeChoose', 'LearnCode', 'LearnCodeAI', 'AILearnHow']


In [48]:
df_basic[['LearnCodeChoose', 'LearnCode', 'LearnCodeAI', 'AILearnHow']].head()

,LearnCodeChoose,LearnCode,LearnCodeAI,AILearnHow
0,"Yes, I am not new to coding but am learning ne...",Online Courses or Certification (includes all ...,"Yes, I learned how to use AI-enabled tools for...",AI CodeGen tools or AI-enabled apps
1,"Yes, I am not new to coding but am learning ne...",Online Courses or Certification (includes all ...,"Yes, I learned how to use AI-enabled tools for...",AI CodeGen tools or AI-enabled apps
2,"Yes, I am not new to coding but am learning ne...",Online Courses or Certification (includes all ...,"Yes, I learned how to use AI-enabled tools for...",AI CodeGen tools or AI-enabled apps;Technical ...
3,"Yes, I am not new to coding but am learning ne...","Other online resources (e.g. standard search, ...","Yes, I learned how to use AI-enabled tools for...",AI CodeGen tools or AI-enabled apps;Videos (no...
4,"No, I am not new to coding and did not learn n...",NaN,"Yes, I learned how to use AI-enabled tools for...",Technical documentation (is generated for/by t...


In [49]:
# Define the name of the column
learn_col = 'LearnCode'

# We filter those who have chosen online courses, we use 'Online Courses'
online_learners = df_basic[df_basic[learn_col].str.contains('Online Courses', case=False, na=False)]

# We count the number
online_count = online_learners.shape[0]

# We derive the result
print(f"\n The number of respondents who studied through online courses: {online_count}")


 The number of respondents who studied through online courses: 10973


In [50]:
# Looking for a column name that contains the word 'Country'
matching_cols = [col for col in df_basic.columns if 'Country' in col]
print("\n Знайдені схожі колонки:", matching_cols)


 Знайдені схожі колонки: ['Country']


In [51]:
# We will use the ready-made dataset on Python developers from the previous task 5 python_users
# the number of Python users from the previous task
print (f"\n Number of Python users: {python_count}")

df_python = python_users.dropna(subset=['ConvertedCompYearly'])
python_clean = df_python.shape[0]
print (f"\n Number of Python users who reported compensation: {python_clean}")

# We group by countries and count metrics
# Add 'count' to see the size of the sample for each country
country_analysis = df_python.groupby('Country')['ConvertedCompYearly'].agg(['mean', 'median', 'count'])

# Rename for convenience
country_analysis.columns = ['mean ($)', 'median ($)', 'Number of respondents']

# We filter countries with at least 5 developers for statistical reliability
final_table = country_analysis[country_analysis['Number of respondents'] > 5].sort_values(by='median ($)', ascending=False)

# We round and change the type to an integer for beauty
final_table = final_table.round(0).astype(int)

# Output of the result
print("\n Salary analysis of Python developers by country:")
display(final_table.head(15))


 Number of Python users: 18466

 Number of Python users who reported compensation: 12765

 Salary analysis of Python developers by country:


,mean ($),median ($),Number of respondents
Country,,,
United States of America,173299,150000,3126
Israel,135828,142594,104
Switzerland,156457,142592,240
Nomadic,120132,139218,7
Ireland,120524,116015,74
Luxembourg,116015,109054,7
Denmark,115672,98289,120
Australia,118091,97514,307
United Kingdom of Great Britain and Northern Ireland,117662,94958,750


In [52]:
# Since df_python no longer has NaNs in salaries, we just sort it
# Sort by decreasing compensation
df_top_paid = df_python.sort_values(by='ConvertedCompYearly', ascending=False)

# We select the TOP-5 respondents and the required columns
# we add 'Country' out of curiosity
top_5 = df_top_paid[['ConvertedCompYearly', 'EdLevel', 'Country']].head(5)

# We derive the result
print("\n Education level of the 5 highest paid Python developers:")
display(top_5)


 Education level of the 5 highest paid Python developers:


,ConvertedCompYearly,EdLevel,Country
48428,10000000.0,"Bachelor’s degree (B.A., B.S., B.Eng., etc.)",Ukraine
30728,6371285.0,"Professional degree (JD, MD, Ph.D, Ed.D, etc.)",Brazil
40418,6000000.0,Some college/university study without earning ...,Viet Nam
19536,5810837.0,"Professional degree (JD, MD, Ph.D, Ed.D, etc.)",Nigeria
14873,4706124.0,"Secondary school (e.g. American high school, G...",United Kingdom of Great Britain and Northern I...


In [53]:
# We count the total number of respondents in each age category
age_total = df_basic['Age'].value_counts()

# We count the number of Python developers in each age category
# We count the number of Python developers in each age category
age_python = python_users['Age'].value_counts()

# We combine these data into one table
age_categoric = pd.concat([age_total, age_python], axis=1)
age_categoric.columns = ['Total', 'Python_Users']

# We calculate the percentage
age_categoric['(%)'] = (age_categoric['Python_Users'] / age_categoric['Total']) * 100

# We round the result for a better look
age_categoric['(%)'] = age_categoric['(%)'].round(1)

# Sort by percentage to see where Python is most popular
# Or you can sort by index to see ages one by one
age_categoric = age_categoric.sort_values(by='(%)', ascending=False)

# We derive the result
print("\n Popularity of Python among different age groups:")
display(age_categoric[['Total','(%)']])


 Popularity of Python among different age groups:


,Total,(%)
Age,,
18-24 years old,9210,40.0
45-54 years old,6275,38.6
55-64 years old,2626,37.2
25-34 years old,16519,36.9
35-44 years old,13241,36.7
65 years or older,942,31.6
Prefer not to say,378,31.2


In [54]:
# Searching for the column name that contains the word 'Ind'
matching_cols = [col for col in df_basic.columns if 'Ind' in col]
print("\n Found similar columns:", matching_cols)


 Found similar columns: ['Industry']


In [55]:
# We filter remote workers
remote_respond = df_basic[df_basic['RemoteWork'] == 'Remote']

# We find the 75th quantile (the limit above which the top 25% of salaries are located)
q75 = remote_respond['ConvertedCompYearly'].quantile(0.75)
print(f"\n Limit of the 75th quantile (Top-25%): ${q75:,.0f}")

# We choose only those who earn the same or more
top_remote = remote_respond[remote_respond['ConvertedCompYearly'] >= q75]

# We count the number of respondents by industry.
# The value_counts() method is more convenient here, because it immediately sorts from largest to smallest
industry_counts = top_remote['Industry'].value_counts()

# Convert to DataFrame for good output
industry_report = industry_counts.reset_index()
industry_report.columns = ['Industry', 'Number of top specialists']

print("\n The most popular industries among high-paying remote workers:")
display(industry_report.head(10))


 Limit of the 75th quantile (Top-25%): $150,000

 The most popular industries among high-paying remote workers:


,Industry,Number of top specialists
0,Software Development,878
1,Fintech,146
2,Healthcare,131
3,Other:,126
4,"Internet, Telecomm or Information Services",101
5,Banking/Financial Services,61
6,Media & Advertising Services,58
7,Retail and Consumer Services,55
8,Government,53
9,"Transportation, or Supply Chain",45


    Results 
    total number of respondents                                     49191
    full answers of respondents                                         0
    Measures of Central Tendency for Work Experience (WorkExp):

                                                           Mean      13.4 
                                                           Median    10.0
                                                           Mode      10.0
                                                           
    The number of respondents who work remotely:                    10931 
    Percentage of respondents who program in Python:                37.5% 
    The number of respondents who studied through online courses:   10973 
    Number of Python users:                                         18466 
    Number of Python users who reported compensation:               12765

            Salary analysis of Python developers by country:
                                Mean ($)	Median ($)	   Number of 
                                                           respondents
    Country			
    United States of America	  173299    	150000	   3126
    Israel	                      135828    	142594	   104
    Switzerland	                  156457    	142592	   24

    Education level of the 5 highest paid Python developers:
    ConvertedCompYearly	             EdLevel	                                                 Country
    48428	10000000.0	             Bachelor’s degree (B.A., B.S., B.Eng., etc.)	             Ukraine
    30728	 6371285.0	             Professional degree (JD, MD, Ph.D, Ed.D, etc.)	             Brazil
    40418	 6000000.0	             Some college/university study without earning ...	         Viet Nam
    19536	 5810837.0	             Professional degree (JD, MD, Ph.D, Ed.D, etc.)	             Nigeria
    14873	 4706124.0	             Secondary school (e.g. American high school, G...	         United Kingdom of Great Britain                                                                                                    and Northern I...

     Popularity of Python among different age groups:
                   Total	(%)
    Age		
    18-24 years old	    9210	40.0
    45-54 years old	    6275	38.6
    55-64 years old	    2626	37.2
    25-34 years old	    16519	36.9
    35-44 years old	    13241	36.7
    65 years or older	 942	31.6
    Prefer not to say	 378	31.2

     Limit of the 75th quantile (Top-25%): $150,000

    The most popular industries among high-paid remote workers:
                Industry Number of top specialists
    0	Software Development	878
    1	Fintech	                146
    2	Healthcare	            131

    Conclusions:
It follows from the review that the Python programming language is quite common, occupying 37.5% of respondents and especially 40% among the youngest age group.
The respondents' average work experience of 13.4 years also indicates a significant share of Python usage among experienced users.
Among the top countries in terms of compensation is the USA, which is also the leader in the number of Python users 3126 - 6.35% of the total number
And the top industry where Python is used is software development with a fairly significant gap from the other two advanced industries of fintech and health care, which is also quite telling. 
So, Python can be considered a top programming language, which is highly valued in developed countries, as evidenced by the level of compensation.